Automattic/wp-calypso
# Design picker: Fix overflowing sidebar

Follow-up to https://github.com/Automattic/wp-calypso/pull/81137

## Proposed Changes

Prevents the Design Picker sidebar content from overflowing horizontally when using overlay scrollbars (i.e. when setting "Show scroll bars" to "When scrolling"):

Show scroll bars | Vertical overflow | Before | After
--- | --- | --- | ---
When scrolling | No | <img width="334" alt="Screenshot 2023-08-30 at 13 05 21" src="https://github.com/Automattic/wp-calypso/assets/1233880/41789c1f-b1ff-465c-aa49-1cc23d4640f8"> | <img width="315" alt="Screenshot 2023-08-30 at 13 05 31" src="https://github.com/Automattic/wp-calypso/assets/1233880/50bf3a51-34d3-4c39-8c62-e52e48d7e1f7">
When scrolling | Yes | <img width="325" alt="Screenshot 2023-08-30 at 13 05 52" src="https://github.com/Automattic/wp-calypso/assets/1233880/f5af759e-5313-4b74-a1d1-35d52c086d2f"> | <img width="317" alt="Screenshot 2023-08-30 at 13 05 57" src="https://github.com/Automattic/wp-calypso/assets/1233880/b4cc5c12-3a3e-4cde-b529-4e50152ff4f1">

The downside is that when the scroll bars are always visible, the content will be "pushed" to the left:

Show scroll bars | Vertical overflow | Before | After
--- | --- | --- | ---
Always | No | <img width="318" alt="always - visible - before" src="https://github.com/Automattic/wp-calypso/assets/1233880/180840cb-e451-408d-b881-e85f6e6674cd"> | <img width="318" alt="always - visible - after" src="https://github.com/Automattic/wp-calypso/assets/1233880/becdfc8f-e4f3-4410-a15c-b7de77b68093">
Always | Yes | <img width="330" alt="always - collapsed - before" src="https://github.com/Automattic/wp-calypso/assets/1233880/ed370fb6-fed3-4693-871a-cf6fdb3cf407"> | <img width="317" alt="Screenshot 2023-08-30 at 13 05 05" src="https://github.com/Automattic/wp-calypso/assets/1233880/d1fd64d2-1738-4c62-ab2b-4fffdd65fde2">

But I don't think we have any way to avoid this. Also, it's something we already do in the Site Assembler anyway, so it's probably not a big deal?

<img width="500" alt="Screenshot 2023-08-30 at 13 02 31" src="https://github.com/Automattic/wp-calypso/assets/1233880/0b74a3c6-9b66-4b99-a387-f5cc0a35eb42">

## Testing Instructions

- Open the Appearance macOS settings and change "Show scroll bars" to "When scrolling"
- Use the Calypso live link below
- Go to `/setup/site-setup/designSetup?siteSlug=<SITE_DOMAIN>&theme=twentytwentythree`.
- Resize the browser so all style variations are visible.
- Make sure the sidebar content doesn't not overflow to the right
- Slightly reduce the browser height so the style variations start to overflow vertically.
- Make sure the sidebar content doesn't not overflow to the right
- Change the "Show scroll bars" setting to "Always now" and repeat the same steps above
